hedera-node/docs/design/services/smart-contract-service/cancun-fork-support.md

```diff
@@ -0,0 +1,111 @@
+  available in the Cancun fork.
+* As a Hedera developer, I want to preserve maximum future design space to adopt, adopt, or not adopt blobs.
+* As an end user, I want prompt and accurate failures if I attempt to use Blob features in Hedera.
+
+## Goals
+
+The principal goal is to maintain EVM and Solidity/Vyper/etc. compatibility, where the bytecode from contracts compiled
+with Cancun enabled smart contract compilers still function correctly.
+
+Where inclusion of features don't require additional work, they too should be included.
+
+Where we have existing overrides, opcodes that change those behaviors should still operate in a fashion that is in
+harmony with the EVM equivalence requirements and the Hedera requirements to the extent possible. Working with Hedera
+takes priority when there is unavoidable conflict.
+
+Opcodes that interact with blobs should not cause the contracts to break, but should not require blobs to work.
+
+## Non Goals
+
+It is not a goal to introduce blobs into Hedera. It is also not a goal to restrict design space nor to dictate future
+design directions for support or non-support of blobs.
+
+## Implementation
+
+### New Cancun EVM
+
+A new EVM version will be created that will be based off of the Cancun EVM - which is from a Besu
+release `>24.1.2`.  A new enum value, `HederaEvmVersion.VERSION_049` will have the value `v0.49`.  
+
+Setting the `@ConfigProperty` `ContractsConfig.evmVersion` (string value `contracts.evm.version`) 
+to `v0.49` (from `v0.46`) will activate the Cancun EVM.
+
+Simply activating the Cancun EVM - with no changes or overrides - will immediately provide the 
+following features:
+
+* `MCOPY` operation
+* `TLOAD` and `TSTORE` operations
+* `VERSIONEDHASH` and `BLOBBASEFEE` operations that query the `TxValues` Object
+* KZG Precompiled contract
+
+### Set Correct Values in TxValues
+
+There are two new fields in the `TxValues` object that need to be set correctly. We need to ensure
+that `versionedHashes` is set to an empty list and that `blobGasPrice` is set to one. We should be able to do that in all
+EVM versions so setting them unconditionally in the `MessageFrame` construction in `HederaEvmTxProcessor` should be
+sufficient.
+
+<!-- **TODO(Nana): Set them in the EVM versions how? Is there a specific class that needs to be updated or a method that needs to be overridden?** -->
+
+
+### Update HederaSelfDestructOperation behavior
+
+Either with a new class or a class that takes a constructor parameter, update the HederaSelfDestructOperation so that it
+will behave consistently with the EIP-6780 `SELFDESTRUCT` rules.
+
+<!-- **TODO(Nana): Please specify which option the implementation should follow. Feel free to not that the other wasn't chosen for reason X** -->
+
+## Acceptance Tests
+
+Verify the presence of the following features in the new EVM via new smart contracts exercising their features:
+
+* `MCOPY` operation
+* `TLOAD` and `TSTORE` operations
```

Comment:
We have custom overloaded `SLoad` and `SStore` operations.  The main reason is that we want to track the loading and storing of values into and out of storage via sidecars.  Do we not want to do this for `TLoad` and `TStore`?

Reply:
Transient storage is transient.  Disappears at the end of a tx.  Does not change persistent (blockchain) state in any way.

This is not to say it isn't interesting to a developer.  For example, a principal use case is to support reentrancy guards, as you know, something that the lack of causes contract security violations.  But (as of now) visibility into that, via the blockchain/mirror node, is not planned.